docs/proposals/secret-manager-store-api.md

```diff
@@ -0,0 +1,69 @@
+# Secrets Manager Store Proposal
+
+**Author**: Emmanuel Bakare (@emmanuel-kubermatic)
+
+**Status**: Draft proposal; prototype in progress.
+
+## Goals
+The proposed fix for the issue [6671](https://github.com/kubermatic/kubermatic/issues/6671) 
+
+Adds support for vault related read access within the kubermatic source
+
+## Non-Goals
+Adds support for vault access and a central store for accessing data
+
+## Motivation and Background
+
+Currently, there's a user ssh agent which manages credentials via fsnotify events and kubernetes secrets.
+
+This approach is great but adds a limitation where the end user has to manually copy credentials to ssh secrets or add them via the API.
```

Comment:
Actually, the current flow is the following:

1. KKP users add secret with KKP UI or API, and [UserSSHKey](https://github.com/kubermatic/kubermatic/blob/9f2af8e327595efbe90b96fd8f44bd51dcafa7ff/pkg/crd/kubermatic/v1/sshkeys.go#L31-L50) get created on the master cluster
2. [master-controller-manager ensures](https://github.com/kubermatic/kubermatic/blob/9f2af8e327595efbe90b96fd8f44bd51dcafa7ff/pkg/controller/master-controller-manager/usersshkeyssynchronizer/controller.go#L199-L206) that secrets containing the certificates are present in the corresponding seed clusters.
3. [user-cluster-controller-manager](https://github.com/kubermatic/kubermatic/blob/9f2af8e327595efbe90b96fd8f44bd51dcafa7ff/pkg/controller/user-cluster-controller-manager/resources/reconciler.go#L69) sync those secrets from seed to user clusters
4. The agents running on each worker [fetch](https://github.com/kubermatic/kubermatic/blob/9f2af8e327595efbe90b96fd8f44bd51dcafa7ff/pkg/controller/usersshkeysagent/usersshkeys_controller.go#L154-L166) the certs from the secret and update the authorized keys.

The key thing to decide to me is at which stage do where do we want to integrate Vault.

Reply:
I've seen that we do not need access via the master-controller-manager for now as we discussed.

Editing the cloud-init configuration is also not needed as we can create the file via a hostPath mount which would then use the VaultSecret named `vaultsignedca` to create a TrustedCA file in the `/etc/ssh` directory.

I'd update that in the proposal